### PR TITLE
Introduce LoginCredentials domain factory and tests; wire login API

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds domain-level authentication primitives: `LoginCredentials` and `LoginValidationError` under `backend/src/domain/auth.rs`.
- Exposes credentials via `domain::auth` and wires a domain factory into the login API so inbound payloads are validated inside the domain.
- Includes tests for domain validation and API-level test for invalid login payloads.

## Changes
- backend/src/domain/auth.rs (new)
  - Implements `LoginCredentials` with validated inputs via `try_from_parts`.
  - Validation rules:
    - Username must be non-empty after trimming (empty/whitespace -> `EmptyUsername`).
    - Password must be non-empty (empty -> `EmptyPassword`).
  - Stores password using `Zeroizing<String>` to mitigate leakage.
  - Accessors: `username()` and `password()`.
  - Domain error: `LoginValidationError` with Display and Error impls.
  - Domain-level unit tests for validation logic (empty/whitespace username, trimmed username, empty password).

- backend/src/domain/mod.rs
  - Exposes: `LoginCredentials`, `LoginValidationError` by importing from `auth`.
  - Declares `pub mod auth;`.

- backend/src/api/users.rs
  - Imports `LoginCredentials` and `LoginValidationError`.
  - Adds `impl LoginCredentials::try_from_login_payload(payload: LoginRequest)` to convert inbound login payload into validated credentials.
  - `login` handler now uses the domain factory to validate input and maps domain errors to API errors (`Error::invalid_request`).
  - Added test: `login_rejects_invalid_payload` to assert BAD_REQUEST for invalid payloads.

- docs/backend-roadmap.md
  - Marked the domain-factory replacement for DTOs as completed.

- docs/wildside-backend-architecture.md
  - Documented `LoginCredentials` as a domain factory that trims usernames and zeroizes passwords; explains design rationale and future wiring to a real `UserRepository`.

- packages/types/package.json
  - Added `typescript` as a dev dependency to support domain-level tests.

- pnpm-lock.yaml
  - Updated to reflect new TypeScript dev dependency.

## Test plan
- [x] Domain tests for `LoginCredentials` validation (empty/whitespace username, trimmed username, empty password).
- [x] API-level test: `login` rejects invalid payload with HTTP 400 BAD_REQUEST.
- [x] Existing tests for user listing remain unaffected.

## How to run
- Backend tests: cargo test -p backend
- If running locally for API tests, ensure Actix test environment is configured (session store, etc.).

🌿 Generated by [Terry](https://www.terragonlabs.com)

---

ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/77c6cc55-d810-476a-a7bc-c21fefa6351b

## Summary by Sourcery

Introduce a LoginCredentials domain factory with validation and wire it into the login API while updating tests and documentation.

New Features:
- Add LoginCredentials and LoginValidationError in the domain layer to encapsulate validated username/password inputs.
- Wire LoginCredentials::try_from_login_payload into the login handler to enforce domain validation of incoming login requests.

Enhancements:
- Trim usernames, reject blank inputs, and zeroize passwords using the zeroize crate.
- Expose LoginCredentials and LoginValidationError in the domain module for adapters to consume.

Build:
- Add TypeScript as a dev dependency in packages/types and update the pnpm lock file.

Documentation:
- Document the LoginCredentials domain factory pattern in the architecture guide.
- Mark the domain-factory replacement task as completed in the backend roadmap.

Tests:
- Add domain unit tests for credential validation edge cases.
- Add an API-level test to ensure invalid login payloads return HTTP 400 BAD_REQUEST.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Mapping LoginValidationError to Error::invalid_request via err.to_string loses structured error codes—consider mapping each variant to a stable error code or ErrorCode instead of just a string.
- The try_from_login_payload impl is currently in the API module; consider moving that constructor into the domain crate so the adapter remains thin and all validation lives in one place.
- The API-level test covers invalid usernames but not empty passwords—add a test case for missing or empty password payloads to ensure the endpoint rejects those as well.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

- Consider implementing TryFrom<LoginRequest> for LoginCredentials instead of a custom try_from_login_payload to leverage the standard conversion trait and improve ergonomics.
- Add a positive API-level test to verify that valid credentials result in a 200 OK response so you cover the successful login path end-to-end.
- Map each LoginValidationError variant to a distinct API error code or message rather than a generic invalid_request to provide clients more precise feedback on why their payload was rejected.

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?



    Error mapping structure ❌ Not resolved
        Line 48 in backend/src/api/users.rs still uses .map_err(|err| Error::invalid_request(err.to_string()))
        This loses the structured error variants (EmptyUsername vs EmptyPassword)
        Consider matching on the error variants to provide stable error codes

    Method location ❌ Not resolved
        try_from_login_payload is still implemented in the API module (lines 26-30 of backend/src/api/users.rs) as an inherent impl on LoginCredentials
        It should be moved to backend/src/domain/auth.rs to keep the adapter thin
        The API layer would then just call domain methods without extending domain types

    Test coverage ❌ Not resolved
        The login_rejects_invalid_payload test (line 116) only covers whitespace username ("   ")
        No test case exists for empty or missing password
        Add a second test case with empty password to ensure both validation paths are covered at the API level

File: docs/wildside-backend-architecture.md [hunk 1]
```diff
@@ -554,6 +554,18 @@ before invoking a port. Canonical examples include:
 - `CacheKey` — encapsulates the canonical hash input for caching so that the
   service, rather than the handler, is responsible for namespacing and TTL
   rules.
+- `LoginCredentials` — trims usernames, zeroises passwords via the `zeroize`
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/wildside-backend-architecture.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.
